prober/http.go

```diff
@@ -123,30 +141,76 @@ func (h *HTTP) calculateInterval() time.Duration {
 }
 
 func (h *HTTP) sendRequest(ctx context.Context) HTTPResult {
-	req, _ := http.NewRequestWithContext(ctx, http.MethodGet, h.URL, nil)
-	if h.Host != "" {
-		req.Host = h.Host
+	var startTime time.Time
+	var finishTime time.Time
+	var dnsStartTime time.Time
+	var dnsDoneTime time.Time
+	var connectStartTime time.Time
+	var connectDoneTime time.Time
+	var tlsHandshakeStartTime time.Time
+	var tlsHandshakeDoneTime time.Time
+	httpTrace := &httptrace.ClientTrace{
+		GetConn: func(_ string) {
+		},
+		GotConn: func(info httptrace.GotConnInfo) {
+			log.Printf("conn was reused: %t", info.Reused)
+
+		},
+		DNSStart: func(_ httptrace.DNSStartInfo) {
+			dnsStartTime = time.Now()
+		},
+		DNSDone: func(_ httptrace.DNSDoneInfo) {
+			dnsDoneTime = time.Now()
```

Comment:
please check DNS errors and add required metrics

Reply:
Also, log the errors with the required metadata

Reply:
done